js-apps/meep-frontend/src/js/util/scenario-utils.js

```diff
@@ -236,25 +237,91 @@ export function parseScenario(scenario) {
   return { table: table, visData: visData };
 }
 
+function findIdInScenario(scenario, uniqueId) {
```

Comment:
Is there a better way of determining if the ID is in the scenario? E.g. by keeping a map of IDs.

Reply:
Only invoked when a new element scenario is created which should happen once and then not anymore. Although the looping through the scenario might seem extensive, checking the scenario for duplicated names is a security of its uniqueness. Keeping a global map of what ids are in the scenario could be useful, but not believed to be essential.